envoy/router/router.h

```diff
@@ -1068,6 +1058,13 @@ class Route {
    */
   virtual const RouteSpecificFilterConfig* perFilterConfig(const std::string& name) const PURE;
 
+  /**
+   * This is a helper to get the route's per-filter config if it exists, otherwise the virtual
+   * host's. Or nullptr if none of them exist.
+   */
+  virtual const RouteSpecificFilterConfig*
+  mostSpecificPerFilterConfig(const std::string& name) const PURE;
```

Comment:
Is there any reason at this point to keep `perFilterConfig` at the interface level? Can we remove it? That can be an internal implementation detail. It seems that users should never care about whether the config comes from route/vhost/etc.?

Reply:
looks like `traversePerFilterConfig` is only place using the `perFilterConfig` out side the class.
https://github.com/envoyproxy/envoy/blob/6834081f835d609606e77ef84d734e30fc403d58/source/common/http/utility.h#L536

So we can move the `traversePerFilterConfig` to the interface also.

Both `traversePerFilterConfig` and `mostSpecificPerFilterConfig` needs the `perFilterConfig`, we can make it as private.